R/MeasureSurvRCLL.R

```diff
@@ -73,16 +73,17 @@ MeasureSurvRCLL = R6::R6Class("MeasureSurvRCLL",
       event = truth[, 2] == 1
       event_times = truth[event, 1]
       cens_times = truth[!event, 1]
+      distr = prediction$distr
 
       if (!any(event)) { # all censored
         # survival at outcome time (survived *at least* this long)
-        out[!event] = diag(as.matrix(prediction$distr[!event]$survival(cens_times)))
+        out[!event] = diag(as.matrix(distr[!event]$survival(cens_times)))
       } else if (all(event)) { # all uncensored
         # pdf at outcome time (survived *this* long)
-        out[event] = diag(as.matrix(prediction$distr[event]$pdf(event_times)))
+        out[event] = diag(as.matrix(distr[event]$pdf(event_times)))
       } else { # mix
-        out[event] = diag(as.matrix(prediction$distr[event]$pdf(event_times)))
-        out[!event] = diag(as.matrix(prediction$distr[!event]$survival(cens_times)))
+        out[event] = diag(as.matrix(distr[event]$pdf(event_times)))
+        out[!event] = diag(as.matrix(distr[!event]$survival(cens_times)))
       }
```

Comment:
I have no problem with this at all but how does it speed up RCLL?

Reply:
In most test sets, there are going to be some censored observations and some uncensored. Then we are calling `prediction$distr` field twice in lines 84, 85, which in case of a `Arrdist` with dim = 75 x 75 x 1000 for example, each call takes already a long time (some secs). Unless there is some caching going on, which I am not aware of, this should speed it up a bit. If not, it's nice refactoring :) 

In general I have seen that RCLL, even though easy to code, is slower than graf which is a much more complicated measure, but I guess this has to do with `distr6` and `R6` vs C++ code?

Reply:
ahh sorry ofc I was forgetting that prediction$distr is an active binding so computations are going on behind the scenes, thanks! and yes it'll be because of several calls to distr6 functions whereas the others handle everything using the matrices

Reply:
I was confused by this, because I understood that calling the interface `distr` takes more time, but when the `weighted_survival_score()` is called in some measures we use `prediction$distr` and not `prediction$data$distr` which should just work

Reply:
It's just protection against different internal representations of the distribution, `prediction$distr` is the active binding and `prediction$data$distr` is just a list entry, can't remember why I chose one over the other in individual cases though.....

Reply:
How does that effect parametric models?

Reply:
I tested with `lrn("surv.parametric")`, all the measures return sensible results.